webClient/src/app/editor/code-editor/file-tabs/file-tabs.component.ts

```diff
@@ -38,17 +38,37 @@ export class FileTabsComponent implements OnInit {
     useBothWheelAxes: true
   };
 
-  constructor(@Inject(Angular2InjectionTokens.VIEWPORT_EVENTS) private viewportEvents: Angular2PluginViewportEvents) { }
+  private iterableDiffer;
+
+  constructor(
+    private editorControl: EditorControlService,
+    @Inject(Angular2InjectionTokens.VIEWPORT_EVENTS) private viewportEvents: Angular2PluginViewportEvents,
+    private _iterableDiffers: IterableDiffers) { 
+    this.iterableDiffer = this._iterableDiffers.find([]).create(null);
+  }
 
   ngOnInit() {
     this.viewportEvents.resized.subscribe(()=> {
       this.componentRef.directiveRef.update();
     });
+    this.editorControl.initializedFile.subscribe(() => {
+      this.componentRef.directiveRef.scrollToRight();
+    });
+
+  }
+
+  ngAfterViewChecked() {
+    const changes = this.iterableDiffer.diff(this.data);
+    if (changes) {
+      this.componentRef.directiveRef.scrollToRight();
```

Comment:
This is half right: you scroll to right when opening.
But if I close, although it selects the one all the way to the left (which is fine), it doesn't scroll to the left.
I suggest listening on editorControl for a close event and doing scrollToLeft() if possible.